integration-test/src/test/java/org/ehcache/integration/SerializerTest.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.integration;
+
+import org.ehcache.CacheManager;
+import org.ehcache.CacheManagerBuilder;
+import org.ehcache.config.CacheConfiguration;
+import org.ehcache.config.CacheConfigurationBuilder;
+import org.ehcache.config.ResourcePoolsBuilder;
+import org.ehcache.config.SerializerConfiguration;
+import org.ehcache.config.persistence.CacheManagerPersistenceConfiguration;
+import org.ehcache.config.serializer.DefaultSerializerConfiguration;
+import org.ehcache.config.units.EntryUnit;
+import org.ehcache.config.units.MemoryUnit;
+import org.ehcache.internal.TimeSourceConfiguration;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+/**
+ * @author rism
+ */
+public class SerializerTest {
+  @Rule
+  public TemporaryFolder folder = new TemporaryFolder();
+
+  @SuppressWarnings("unchecked")
+  @Test
+  public void testSerializers() throws Exception {
+    TestTimeSource testTimeSource = new TestTimeSource();    
+    CacheConfigurationBuilder configurationBuilder = CacheConfigurationBuilder.newCacheConfigurationBuilder()
+        .withResourcePools(ResourcePoolsBuilder.newResourcePoolsBuilder()
+            .heap(10, EntryUnit.ENTRIES)
+            .disk(10, MemoryUnit.MB))
+        .add(new DefaultSerializerConfiguration<CharSequence>(TestSerializer.getBuggySerializer(true), SerializerConfiguration.Type.VALUE));
+    CacheConfiguration<Integer, CharSequence> configuration = configurationBuilder.buildConfig(Integer.class, CharSequence.class);
+    CacheManager cacheManager = CacheManagerBuilder.newCacheManagerBuilder().withCache("myCache", configuration)
+        .using(new TimeSourceConfiguration(testTimeSource))
+        .with(new CacheManagerPersistenceConfiguration(folder.newFolder("tempData")))
+        .build(false);
+    try {
+      cacheManager.init();
+    } catch (Exception e) {
```

Comment:
Can't you be more specific as to what you catch?
Also, if no exception is thrown, the test will proceed fine which is not what we want ...